### ISSUE TITLE
transforms.RandomRotation with Tensors leads to an error

### ISSUE THREAD

**Issue by tonystratum** [pytorch/vision#3309]
## 🐛 Bug

Originally (vision/nightly), when `torchvision.transforms.RandomRotation` is used with a Tensor of shape `(C, H, W)`, without a `fill` argument, we get a following error:

<details>
  <summary>Error</summary>
 
  
  ```python
Traceback (most recent call last):
  File "repr1.py", line 9, in <module>
    tensor = torchvision.transforms.RandomRotation(45)(tensor)
  File "/home/stratum/miniconda3/envs/pytorch-cpu-nightly/lib/python3.6/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/stratum/miniconda3/envs/pytorch-cpu-nightly/lib/python3.6/site-packages/torchvision/transforms/transforms.py", line 1236, in forward
    fill = [float(f) for f in fill]
TypeError: 'NoneType' object is not iterable
  ```
  
</details>

## To Reproduce

```python
import torch
import torchvision
tensor = torch.rand(3, 224, 224)
tensor = torchvision.transforms.RandomRotation(45)(tensor)
print(tensor.shape)
```

or

```python
import torch
import torchvision
tensor = torch.rand(3, 224, 224)
transforms = torch.nn.Sequential(
    torchvision.transforms.RandomRotation(45)
)
transformed = transforms(tensor)
print(tensor.shape)
```

## Expected behavior

```python
torch.Size([3, 224, 224])
```

## Environment

<details>
  <summary>Environment</summary>
 
  
  ```python
Collecting environment information...
PyTorch version: 1.8.0.dev20210124
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Manjaro Linux (x86_64)
GCC version: (GCC) 10.2.0
Clang version: Could not collect
CMake version: version 3.19.3

Python version: 3.6 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1660 Ti
Nvidia driver version: 460.32.03
cuDNN version: Probably one of the following:
/usr/lib/libcudnn.so.8.0.5
/usr/lib/libcudnn_adv_infer.so.8.0.5
/usr/lib/libcudnn_adv_train.so.8.0.5
/usr/lib/libcudnn_cnn_infer.so.8.0.5
/usr/lib/libcudnn_cnn_train.so.8.0.5
/usr/lib/libcudnn_ops_infer.so.8.0.5
/usr/lib/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.8.0.dev20210124
[pip3] torchaudio==0.8.0a0+e38b122
[pip3] torchvision==0.9.0.dev20210124
[conda] blas                      1.0                         mkl
[conda] cpuonly                   1.0                           0    pytorch-nightly
[conda] mkl                       2020.2                      256
[conda] mkl-service               2.3.0            py36he8ac12f_0
[conda] mkl_fft                   1.2.0            py36h23d657b_0
[conda] mkl_random                1.1.1            py36h0573a6f_0
[conda] numpy                     1.19.2           py36h54aff64_0
[conda] numpy-base                1.19.2           py36hfa32c7d_0
[conda] pytorch                   1.8.0.dev20210124     py3.6_cpu_0  [cpuonly]  pytorch-nightly
[conda] torchaudio                0.8.0.dev20210124            py36    pytorch-nightly
[conda] torchvision               0.9.0.dev20210124        py36_cpu  [cpuonly]  pytorch-nightly
  ```
  
</details>

## Additional context

I proposed a fix in #3303 

Hope this helps!

cc @vfdev-5

**Comment by vfdev-5** [comment]
Thanks for the report @tonystratum ! Looks like it is a bug introduced recently as stable version works.

cc @voldemortX maybe it could be related to your PR : https://github.com/pytorch/vision/pull/2904
could you please take a look ?

**Comment by voldemortX** [comment]
@vfdev-5 Yes, I think I'm to blame for this bug #2904 . @tonystratum Can you check if the same bug exists for other affine transformations, `RandomAffine` and `RandomPerspective` also?

**Comment by vfdev-5** [comment]
@voldemortX yes, same for `RandomAffine`:
```python
tensor = torchvision.transforms.RandomAffine(45, fill=None)(tensor)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/test/lib/python3.6/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/opt/conda/envs/test/lib/python3.6/site-packages/torchvision/transforms/transforms.py", line 1394, in forward
    fill = [float(f) for f in fill]
TypeError: 'NoneType' object is not iterable
```

**Comment by tonystratum** [comment]
> `RandomPerspective`

```python
Traceback (most recent call last):
  File "repr2.py", line 10, in <module>
    transformed = transforms(tensor)
  File "/home/stratum/miniconda3/envs/pytorch-cpu-nightly/lib/python3.6/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/stratum/miniconda3/envs/pytorch-cpu-nightly/lib/python3.6/site-packages/torch/nn/modules/container.py", line 119, in forward
    input = module(input)
  File "/home/stratum/miniconda3/envs/pytorch-cpu-nightly/lib/python3.6/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/stratum/miniconda3/envs/pytorch-cpu-nightly/lib/python3.6/site-packages/torchvision/transforms/transforms.py", line 707, in forward
    fill = [float(f) for f in fill]
TypeError: 'NoneType' object is not iterable
```

**Comment by voldemortX** [comment]
I'd recommend the same fix applied for all 3 affine transform classes since we indeed support fill as an Optional arg. Also add some tests for None at `test/test_transforms_tensor.py`, for instance here:
https://github.com/pytorch/vision/blob/d732562f30b8a83af9ab074592af3dd01fe708ee/test/test_transforms_tensor.py#L371

Sorry for creating this bug... And thanks for catching it @tonystratum

**Comment by tonystratum** [comment]
@voldemortX test for a case when fill is passed with None? (`RandomPerspective(fill=None)`)

**Comment by voldemortX** [comment]
> @voldemortX test for a case when fill is passed with None? (`RandomPerspective(fill=None)`)

On second thought,
if set default to 0, then maybe other than this only the documentation needs changing? @vfdev-5

**Comment by vfdev-5** [comment]
> > @voldemortX test for a case when fill is passed with None? (`RandomPerspective(fill=None)`)
> 
> On second thought,
> if set default to 0, then maybe other than this only the documentation needs changing? @vfdev-5

And ensure that fill is not optional anymore and is not None. As default becomes 0, it is a sort of breaking change as there may be some folks who set it exlicitly as None.

As for `fillcolor`, it is a deprecated arg and should remain None.

**Comment by tonystratum** [comment]
> As for `fillcolor`, it is a deprecated arg and should remain None. (see #2479 for details)

The type check is present

https://github.com/pytorch/vision/blob/cae5400db5d51fc30beea6285e609fcf65f43bd3/torchvision/transforms/transforms.py#L1314

Also, I have just noticed that `interpolation` in `RandomRotation` has a default value
https://github.com/pytorch/vision/blob/cae5400db5d51fc30beea6285e609fcf65f43bd3/torchvision/transforms/transforms.py#L1189

But the docs say that is isn't
https://github.com/pytorch/vision/blob/cae5400db5d51fc30beea6285e609fcf65f43bd3/torchvision/transforms/transforms.py#L1168

Other transformations may be the same, will take a look
cc @vfdev-5

**Comment by vfdev-5** [comment]
Close as fixed in #3303
